Brainpool curves are not supported in Java 17

## Expected Behavior

I expected all unit tests to run correctly under Java 17.

## Actual Behavior

Unit tests that use brainpool curves fail.

## Steps to Reproduce the Problem

```
master * % java -version
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+10 (build 17.0.6+10)
OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (build 17.0.6+10, mixed mode)

master * % ./gradlew test

Welcome to Gradle 7.5!

Here are the highlights of this release:
 - Support for Java 18
 - Support for building with Groovy 4
 - Much more responsive continuous builds
 - Improved diagnostics for dependency resolution

For more details see https://docs.gradle.org/7.5/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

> Task :identity:compileJava
Note: /Volumes/Projects/DVV/identity-credential/identity/src/main/java/com/android/identity/internal/Util.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :identity-android:compileDebugJavaWithJavac
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :identity-android:compileReleaseJavaWithJavac
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :identity:test

com.android.identity.internal.UtilTest > coseSignAndVerify_brainpoolP256r1 FAILED
    java.lang.IllegalStateException at UtilTest.java:471
        Caused by: java.security.SignatureException at UtilTest.java:471

com.android.identity.internal.UtilTest > coseSignAndVerify_brainpoolP384r1 FAILED
    java.lang.IllegalStateException at UtilTest.java:494
        Caused by: java.security.SignatureException at UtilTest.java:494

com.android.identity.internal.UtilTest > coseSignAndVerify_brainpoolP512r1 FAILED
    java.lang.IllegalStateException at UtilTest.java:517
        Caused by: java.security.SignatureException at UtilTest.java:517

com.android.identity.keystore.BouncyCastleKeystoreTest > testEcKeySigningAllCurves FAILED
    org.bouncycastle.operator.RuntimeOperatorException at BouncyCastleKeystoreTest.java:492
        Caused by: java.security.SignatureException at BouncyCastleKeystoreTest.java:492

com.android.identity.keystore.BouncyCastleKeystoreTest > testEcKeyEcdhAllCurves FAILED
    java.lang.AssertionError at BouncyCastleKeystoreTest.java:640
        Caused by: java.security.InvalidAlgorithmParameterException at BouncyCastleKeystoreTest.java:617

com.android.identity.mdoc.request.DeviceRequestParserTest > testDeviceRequestParserReaderAuth_brainpoolP256r1 FAILED
    java.lang.IllegalStateException at DeviceRequestParserTest.java:243
        Caused by: java.security.SignatureException at DeviceRequestParserTest.java:243

com.android.identity.mdoc.request.DeviceRequestParserTest > testDeviceRequestParserReaderAuth_brainpoolP384r1 FAILED
    java.lang.IllegalStateException at DeviceRequestParserTest.java:243
        Caused by: java.security.SignatureException at DeviceRequestParserTest.java:243

com.android.identity.mdoc.request.DeviceRequestParserTest > testDeviceRequestParserReaderAuth_brainpoolP512r1 FAILED
    java.lang.IllegalStateException at DeviceRequestParserTest.java:243
        Caused by: java.security.SignatureException at DeviceRequestParserTest.java:243

com.android.identity.mdoc.sessionencryption.SessionEncryptionTest > testBrainpool256 FAILED
    java.lang.IllegalStateException at SessionEncryptionTest.java:161
        Caused by: java.security.InvalidAlgorithmParameterException at SessionEncryptionTest.java:161

com.android.identity.mdoc.sessionencryption.SessionEncryptionTest > testBrainpool320 FAILED
    java.lang.IllegalStateException at SessionEncryptionTest.java:161
        Caused by: java.security.InvalidAlgorithmParameterException at SessionEncryptionTest.java:161

com.android.identity.mdoc.sessionencryption.SessionEncryptionTest > testBrainpool384 FAILED
    java.lang.IllegalStateException at SessionEncryptionTest.java:161
        Caused by: java.security.InvalidAlgorithmParameterException at SessionEncryptionTest.java:161

com.android.identity.mdoc.sessionencryption.SessionEncryptionTest > testBrainpool521 FAILED
    java.lang.IllegalStateException at SessionEncryptionTest.java:161
        Caused by: java.security.InvalidAlgorithmParameterException at SessionEncryptionTest.java:161

168 tests completed, 12 failed, 4 skipped

> Task :identity:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':identity:test'.
> There were failing tests. See the report at: file:///Volumes/Projects/DVV/identity-credential/identity/build/reports/tests/test/index.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 45s
111 actionable tasks: 111 executed

```

--- a/identity/build.gradle
+++ b/identity/build.gradle
@@ -18,9 +18,9 @@ java {
 dependencies {
     implementation("androidx.annotation:annotation:1.5.0")
     implementation "co.nstant.in:cbor:0.9"
-    implementation "org.bouncycastle:bcprov-jdk15on:1.67"
-    implementation("org.bouncycastle:bcpkix-jdk15on:1.67")
+    implementation "org.bouncycastle:bcprov-jdk15on:1.69"
+    implementation("org.bouncycastle:bcpkix-jdk15on:1.69")
 
     testImplementation "junit:junit:4.13.2"
-    testImplementation "org.bouncycastle:bcprov-jdk15on:1.67"
+    testImplementation "org.bouncycastle:bcprov-jdk15on:1.69"
 }

--- a/identity/src/test/java/com/android/identity/TestUtilities.java
+++ b/identity/src/test/java/com/android/identity/TestUtilities.java
@@ -102,7 +102,7 @@ public class TestUtilities {
         }
 
         try {
-            KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC");
+            KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC", new BouncyCastleProvider());
             ECGenParameterSpec ecSpec = new ECGenParameterSpec(stdName);
             kpg.initialize(ecSpec);
             KeyPair keyPair = kpg.generateKeyPair();

--- a/identity/src/test/java/com/android/identity/internal/UtilTest.java
+++ b/identity/src/test/java/com/android/identity/internal/UtilTest.java
@@ -76,7 +76,7 @@ public class UtilTest {
 
     @Before
     public void setUp() {
-        Security.addProvider(new BouncyCastleProvider());
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
     }
 
     @After

--- a/identity/src/test/java/com/android/identity/keystore/BouncyCastleKeystoreTest.java
+++ b/identity/src/test/java/com/android/identity/keystore/BouncyCastleKeystoreTest.java
@@ -45,7 +45,7 @@ public class BouncyCastleKeystoreTest {
 
     @Before
     public void setup() {
-        Security.addProvider(new BouncyCastleProvider());
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
     }
 
     @Test

--- a/identity/src/test/java/com/android/identity/mdoc/request/DeviceRequestParserTest.java
+++ b/identity/src/test/java/com/android/identity/mdoc/request/DeviceRequestParserTest.java
@@ -56,7 +56,7 @@ public class DeviceRequestParserTest {
 
     @Before
     public void setUp() {
-        Security.addProvider(new BouncyCastleProvider());
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
     }
 
     @After

--- a/identity/src/test/java/com/android/identity/mdoc/sessionencryption/SessionEncryptionTest.java
+++ b/identity/src/test/java/com/android/identity/mdoc/sessionencryption/SessionEncryptionTest.java
@@ -22,7 +22,10 @@ import com.android.identity.internal.Util;
 import com.android.identity.mdoc.engagement.EngagementParser;
 import com.android.identity.util.Constants;
 
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Test;
 
@@ -30,6 +33,7 @@ import java.math.BigInteger;
 import java.security.KeyPair;
 import java.security.PrivateKey;
 import java.security.PublicKey;
+import java.security.Security;
 import java.util.OptionalLong;
 
 import co.nstant.in.cbor.model.Array;
@@ -37,6 +41,16 @@ import co.nstant.in.cbor.model.ByteString;
 import co.nstant.in.cbor.model.DataItem;
 
 public class SessionEncryptionTest {
+
+    @Before
+    public void setUp() {
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
+    }
+
+    @After
+    public void tearDown() {
+        Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
+    }
 
     @Test
     public void testReaderAgainstVectors() {

